src/evict/evict_page.c

```diff
@@ -677,6 +677,104 @@ __evict_child_check(WT_SESSION_IMPL *session, WT_REF *parent)
     return (0);
 }
 
+/*
+ * __evict_review_obsolete_time_window --
+ *     Check whether the ref has obsolete time window information and mark it for dirty eviction to
+ *     remove those obsolete data. An exclusive lock on the page has already been obtained by the
+ *     caller.
+ */
+static int
+__evict_review_obsolete_time_window(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+    WT_ADDR_COPY addr;
+    WT_MULTI *multi;
+    WT_PAGE_MODIFY *mod;
+    WT_TIME_AGGREGATE newest_ta;
+    uint32_t i;
+    char time_string[WT_TIME_STRING_SIZE];
+
+    /*
+     * Pages that the application threads are evicting should not be included. Reconciliation must
+     * be performed when converting a clean page to a dirty page, which could lengthen the
+     * application thread's reconciliation process and increase latency.
+     */
+    if (!F_ISSET(session, WT_SESSION_EVICTION))
+        return (0);
+
+    /* Do not perform any obsolete time window cleanup during the startup or shutdown phase. */
+    if (F_ISSET(S2C(session), WT_CONN_RECOVERING | WT_CONN_CLOSING))
```

Comment:
Why we cannot do that in recovering and shutdown?

Reply:
I think this is to keep those two stages quick. I wonder if we should use `WT_CONN_MINIMAL` as well.

Reply:
Yes, to speed up those two scenarios. 
The `WT_CONN_MINIMAL` has more functions covering the flow of recovery, whereas it covers less part of the shutdown flow. The pages of the table are not loaded until recovery and we do not want any page to be dirtied after the final checkpoint. So I go with `WT_CONN_RECOVERING` and `WT_CONN_CLOSING_CHECKPOINT`.